### PR TITLE
Decompile battle_setup.s

### PR DESCRIPTION
**Pull request by PikalaxALT**

### CODE REVIEW

File: src/battle/battle_setup.c [hunk 1]
```diff
@@ -0,0 +1,575 @@
...
+    setup->safariBalls = balls;
+    return setup;
+}
+
+BattleSetup* BattleSetup_New_Tutorial(HeapID heapId, FieldSystem* fieldSystem) {
+    PlayerProfile* profile = Save_PlayerData_GetProfileAddr(fieldSystem->saveData);
+    OPTIONS* options = Save_PlayerData_GetOptionsAddr(fieldSystem->saveData);
+    BattleSetup* setup = BattleSetup_New(heapId, BATTLE_TYPE_TUTORIAL);
+    setup->saveData = fieldSystem->saveData;
+    {
```
Comment (adrienntindall) [inline src/battle/battle_setup.c:115]: what's up with the weird brackets?

File: src/battle/battle_setup.c [hunk 2]
```diff
@@ -0,0 +1,575 @@
...
+    FreeToHeap(setup->bag);
+    FreeToHeap(setup->pokedex);
+    FreeToHeap(setup->options);
+    sub_02067A78(setup->unk_134);
+    FreeToHeap(setup->bugContestMon);
+    FreeToHeap(setup);
+}
+
+void BattleSetup_AddMonToParty(BattleSetup* setup, Pokemon* mon, int battlerId) {
+    GF_ASSERT(battlerId < 4);
```
Comment (adrienntindall) [inline src/battle/battle_setup.c:175]: Add a `BATTLER_MAX` variable to `constants/battle.h`, then `battlerId < BATTLER_MAX`

File: src/battle/battle_setup.c [hunk 3]
```diff
@@ -0,0 +1,575 @@
...
+    PlayerProfile_Copy(profile, setup->profile[battlerId]);
+}
+
+static void BattleSetup_SetChatotVoiceClip(BattleSetup* setup, SOUND_CHATOT* chatot, int battlerId) {
+    Chatot_Copy(setup->chatot[battlerId], chatot);
+}
+
+void sub_02051D18(BattleSetup* setup, FieldSystem* fieldSystem, SaveData* saveData, u32 mapno, void* arg4, void* arg5) {
+    PlayerProfile* profile; // sp1C
```
Comment (adrienntindall) [inline src/battle/battle_setup.c:194]: remove decomp comments

File: src/battle/battle_setup.c [hunk 4]
```diff
@@ -0,0 +1,575 @@
...
+    profile = Save_PlayerData_GetProfileAddr(saveData);
+    party = SaveArray_Party_Get(saveData);
+    bag = Save_Bag_Get(saveData);
+    pokedex = Save_Pokedex_Get(saveData);
+    chatot = Save_Chatot_Get(saveData);
+    options = Save_PlayerData_GetOptionsAddr(saveData);
+    local = Save_LocalFieldData_Get(saveData);
+
+    if (fieldSystem != NULL) {
+        forceNite = FALSE;
```
Comment (adrienntindall) [inline src/battle/battle_setup.c:213]: Nite? Shouln't it be Night? For this and the RTC_TIMEOFDAY_ const?

File: src/battle/battle_setup.c [hunk 5]
```diff
@@ -0,0 +1,575 @@
...
+    setup->saveData = saveData;
+}
+
+void BattleSetup_InitFromFieldSystem(BattleSetup* setup, FieldSystem *fieldSystem) {
+    sub_02051D18(setup, fieldSystem, fieldSystem->saveData, fieldSystem->location->mapId, fieldSystem->unk94, fieldSystem->unkB0);
+    sub_02052580(setup);
+}
+
+void BattleSetup_InitForFixedLevelFacility(BattleSetup* setup, FieldSystem *fieldSystem, int level) {
+    PlayerProfile* profile; // r4
```
Comment (adrienntindall) [inline src/battle/battle_setup.c:268]: Remove decomp comments

File: src/battle/battle_setup.c [hunk 6]
```diff
@@ -0,0 +1,575 @@
...
+    setup->unk_12C = sub_0202CA44(fieldSystem->saveData);
+    setup->gameStats = Save_GameStats_Get(fieldSystem->saveData);
+    setup->palPad = Save_PalPad_Get(fieldSystem->saveData);
+    setup->mapNumber = fieldSystem->location->mapId;
+    setup->saveData = fieldSystem->saveData;
+    sub_02052580(setup);
+}
+
+void sub_020520B0(BattleSetup* setup, FieldSystem *fieldSystem, Party *party, u8 *a4) {
+    PlayerProfile* profile; // sp20
```
Comment (adrienntindall) [inline src/battle/battle_setup.c:316]: Remove decomp comments etc

File: src/battle/battle_setup.c [hunk 7]
```diff
@@ -0,0 +1,575 @@
...
+        }
+    }
+}
+
+void sub_0205239C(BattleSetup* setup, FieldSystem* fieldSystem) {
+    PlayerProfile* profile;
+    Party* party;
+    Bag* bag;
+    Pokedex* pokedex;
+    SaveVarsFlags* vars_flags;
```
Comment (adrienntindall) [inline src/battle/battle_setup.c:428]: change snake_case -> camelCase

**Review by adrienntindall — CHANGES_REQUESTED** [review]

File: src/battle/battle_setup.c (hunk 1 above)
Comment (PikalaxALT) [reply src/battle/battle_setup.c:115]: They're to enforce the scope of msgData and name. After String_Delete, name is invalid and should not be accessed. After DestroyMsgData, msgData is invalid and should not be accessed.

File: src/battle/battle_setup.c (hunk 7 above)
Comment (PikalaxALT) [reply src/battle/battle_setup.c:428]: How about no?

Comment (adrienntindall) [reply src/battle/battle_setup.c:428]: Fine I'll just do it in a later PR if I remember

**Review by adrienntindall — APPROVED** [review]

**Pull request merged**